plutus-core/plutus-ir/src/PlutusIR/Transform/LetFloatIn.hs

```diff
@@ -48,8 +48,24 @@ binding, or a strict binding whose RHS is work-free. Usually there's no need to
 between these two, since `let x (nonstrict) = rhs` is essentially equivalent to
 `let x (strict) = all a rhs`.
 
+-- Floating a `TypeBind` inwards has no impact on cost (since there's no type in UPLC),
+    -- cannot enable other bindings to be floated inwards, and there is no known example
+    -- where it
+
+-------------------------------------------------------------------------------
+2. What about type and datatype bindings?
+-------------------------------------------------------------------------------
+
+There is no reason to float in a `TypeBind`:
+- Doing so has no impact on cost (since there's no type in UPLC);
+- Doing so cannot enable other bindings to be floated inwards;
+- There is no known example where doing so makes another optimization easier.
+  If anything, doing so may negatively impact the inliner.
+
+`DatatypeBind`s can always be floated inwards. A `DatatypeBind` defines ???
+
 -------------------------------------------------------------------------------
-2. The effect of floating in
+3. The effect of floating in
```

Comment:
You could just make these separate notes, that's often what we do. Up to you though.